Code/Framework/AzFramework/AzFramework/Viewport/CameraInput.cpp

```diff
@@ -539,7 +539,8 @@ namespace AzFramework
             if (input->m_state == InputChannel::State::Began)
             {
                 if (auto translation = TranslationFromKey(input->m_channelId, m_translateCameraInputChannelIds);
-                    translation != TranslationType::Nil)
+                    (state.m_modifiers.GetActiveModifierKeys() & ~ModifierKeyMask::ShiftAny) == ModifierKeyMask::None
+                    && translation != TranslationType::Nil)
```

Comment:
The behavior has changed between the legacy ShortcutDispatcher and the new Action Manager.

In the legacy system, a "top-down" approach was used where the ShortcutDispatcher would direct the shortcuts to a different part of the Qt hierarchy. So when `Ctrl+D` was pressed, for example, it would be consumed by the entity system before it would get to the viewport.

With the new Action Manager, we moved to a more "bottom-up" approach that matches the regular Qt behavior. When a shortcut is pressed, it bubbles up from the focused widget to the root of the widget hierarchy until it finds a widget that consumes it.

So the reason we didn't have this issue with the legacy system is that the ShortcutDispatcher would just send the shortcut to the system that would consume it directly, bypassing the viewport. So the viewport didn't have to check whether WASD were pressed alongside a modifier, since shortcuts would be caught earlier. The new system, instead, passes from the viewport before it reaches the system that eats the shortcut, and as such triggers both.

I honestly deem the new behavior to be the correct one (hence the paradigm shift in the first place).

Reply:
Thanks for the detailed feedback @AMZN-daimini and we've also chatted offline for others following this. @AMZN-daimini is going to try a change where a shortcut can be registered with the viewport widget directly, so it receives the shortcut action first, and the key event then doesn't make it to the viewport keyboard input handling code. If it is a giant pain to change this we can defer it but this would be a really lovely addition and still keep the bottom-up approach. Thanks for looking into this!

Reply:
I went for a different fix for the issue, closing this conversation :)